src/lib/metric_ingest.py

```diff
@@ -119,6 +119,25 @@ async def log_invalid_lines(context: MetricsContext, ingest_response_json: Dict,
                 invalid_line_error_message = invalid_line_error_message.get("error", "")
                 context.log(f"INVALID LINE: '{lines_batch[line_index].to_string()}', reason: '{invalid_line_error_message}'")
 
+class DtDimensionsMap:
+    def __init__(self) -> None:
+            self.dt_dimensions_set_by_source_dimension = {}
+
+    def add_label_mapping(self, source_dimension, dt_target_dimension):
+        all_dt_dims_for_source_dim = self.dt_dimensions_set_by_source_dimension.get(source_dimension, set())
+        all_dt_dims_for_source_dim.add(dt_target_dimension)
+        self.dt_dimensions_set_by_source_dimension[source_dimension] = all_dt_dims_for_source_dim
+
+    def create_dimensions(self, target_dimension_list: List, dt_label_if_unmapped: str, source_label: str, dim_value, context):
```

Comment:
I didn't like the output param either, but was scared to touch it at the time, because I didn't have automatic test then, so this was a welcome reminder 👍 

I was targeting brevity on the calling side back then - it's now bit more verbose now, but I'm happy with it as it still seems clear enough after the change - take a look guys if you like it better or have more thoughts about it.